File: backend/src/routes/userRoutes.js
const express = require('express');
const router = express.Router();
const userController = require('../controllers/userController');

// Endpoint temporário para listar todos os usuários
router.get('/all', userController.listAllUsers);

// Rota para criar/atualizar consultoria do usuário
router.post('/set-consultoria', userController.setUserConsultoria);
router.get('/consultoria', userController.getUserConsultoria);
router.get('/scope', userController.getUserScope);
router.post('/create-child', userController.createChildUser);
router.post('/create-grandchild', userController.createGrandchildUser);
router.post('/create-client-login', userController.createClientLogin);
router.get('/managed', userController.listManagedUsers);
router.put('/managed/:id', userController.updateManagedUser);
router.patch('/managed/:id/active', userController.toggleManagedUserActive);
router.delete('/managed/:id', userController.deleteManagedUser);
router.post('/reset-password-neto', userController.resetPasswordNeto);

module.exports = router;

File: python_backend/routes/user_routes.py
from fastapi import APIRouter, Depends, Query
from controllers import user_controller

router = APIRouter()

@router.get('/all')
async def list_all_users():
    return await user_controller.list_all_users()

@router.get('/consultoria')
async def get_consultoria(email: str = Query(...)):
    return await user_controller.get_user_consultoria(email)

@router.post('/set-consultoria')
async def set_consultoria(payload: dict):
    email = payload.get('email')
    consultoria = payload.get('consultoria')
    return await user_controller.set_user_consultoria(email, consultoria)
